src/appointments.ts

```diff
@@ -189,3 +189,187 @@ export async function checkForAppointments(
     return await proceedWithoutACode(page);
   }
 }
+
+export async function bookAppointment(
+page: Page, 
+title: string,
+firstname: string, 
+lastname: string, 
+zip: string, 
+city: string, 
+street: string, 
+streetnumber: string,
+mobile: string,
+email: string,
+earliestdate: string,
+latestdate: string
```

Comment:
Could you extract that information into a separate data interface? That would make the code clearer.

```ts
interface BookingInformation {
  title: string,
  // ...
```
```

Reply:
Hmm, wouldn't it be better to have it as part of every queue entry, because sometimes not all queue entries are for the same person if you try to book appointments for different family members. And since the personal data relates directly to a booking code and url, I would propose to leave this information all together.  Or do you mean to have another array object inside the queue array?

Reply:
yes it should stay as a queue entry but was just an proposal for splitting the code